src/Standards/PEAR/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php

```diff
@@ -121,22 +130,34 @@ public function process(File $phpcsFile, $stackPtr)
                         $foundIndent = 0;
                     }
 
-                    if ($foundIndent !== $requiredIndent) {
+                    $minIndent      = $previousIndent;
+                    $maxIndent      = $previousIndent;
+                    $expectedIndent = $previousIndent;
+
+                    if ($this->multilevel === true) {
+                        $minIndent      = \max(($previousIndent - $this->indent), $baseIndent);
+                        $maxIndent      = ($previousIndent + $this->indent);
+                        $expectedIndent = \min(\max($foundIndent, $minIndent), $maxIndent);
+                    }
+
+                    if ($foundIndent < $minIndent || $foundIndent > $maxIndent) {
                         $error = 'Object operator not indented correctly; expected %s spaces but found %s';
                         $data  = [
-                            $requiredIndent,
+                            $expectedIndent,
                             $foundIndent,
                         ];
 
                         $fix = $phpcsFile->addFixableError($error, $next, 'Incorrect', $data);
                         if ($fix === true) {
-                            $spaces = str_repeat(' ', $requiredIndent);
+                            $spaces = str_repeat(' ', $expectedIndent);
                             if ($foundIndent === 0) {
                                 $phpcsFile->fixer->addContentBefore($next, $spaces);
                             } else {
                                 $phpcsFile->fixer->replaceToken(($next - 1), $spaces);
                             }
                         }
```

Comment:
Just wondering: should `$previousIndent` be (re)set in this leaf of the condition as well ? Maybe to `$expectedIndent` ?

Reply:
This is one of the possible heuristics. 

Notice that it affects error reporting only:

```php
// Keeping the last valid
$builder
    ->start()
            ->do() // Excpected 8, found 12
                ->end(); // Expected 8, found 16

// Keeping the last adjusted
$builder
    ->start()
            ->do() // Excpected 8, found 12
                ->end(); // Expected 12, found 16
```
Let's go with your suggestion as it may lead to more correct results most of the time, assuming that if there are extraneous spaces, it is more likely that there should be indentation than not.